src/main/java/com/openshift/jenkins/plugins/pipeline/model/RetryIClient.java

```diff
@@ -0,0 +1,241 @@
+package com.openshift.jenkins.plugins.pipeline.model;
+
+import hudson.model.TaskListener;
+
+import java.net.URL;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import com.openshift.jenkins.plugins.pipeline.MessageConstants;
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.IOpenShiftWatchListener;
+import com.openshift.restclient.IResourceFactory;
+import com.openshift.restclient.IWatcher;
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.UnsupportedVersionException;
+import com.openshift.restclient.api.ITypeFactory;
+import com.openshift.restclient.authorization.IAuthorizationContext;
+import com.openshift.restclient.capability.CapabilityVisitor;
+import com.openshift.restclient.capability.ICapability;
+import com.openshift.restclient.model.IList;
+import com.openshift.restclient.model.IResource;
+import com.openshift.restclient.model.JSONSerializeable;
+
+public class RetryIClient implements IClient {
+    
+    private static int MAX_RETRY = 3;
+
+    private final IClient client;
+    private final TaskListener listener;
+    
+    private Object retry(Callable<Object> call) throws OpenShiftException {
+        int retryCount = 0;
+        OpenShiftException ose = null;
+        while (retryCount < MAX_RETRY) {
+            try {
+                Object o = call.call();
+                return o;
+            } catch (Throwable t) {
+                ose = new OpenShiftException(t, "retry", call);
+                listener.getLogger().println(String.format(MessageConstants.RETRY, t.getMessage()));
+                retryCount++;
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException e) {
+                    ose = new OpenShiftException(e, "retry", call);
+                }
+            }
+        }
+        listener.getLogger().println(String.format(MessageConstants.GIVE_UP_RETRY, ose.getCause().getMessage()));
+        throw ose;
+    }
+    
+    private void retry(Runnable call) throws OpenShiftException {
+        int retryCount = 0;
+        OpenShiftException ose = null;
+        while (retryCount < MAX_RETRY) {
+            try {
+                call.run();
+            } catch (Throwable t) {
+                ose = new OpenShiftException(t, "retry", call);
+                listener.getLogger().println(String.format(MessageConstants.RETRY, t.getMessage()));
+                retryCount++;
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException e) {
+                    ose = new OpenShiftException(e, "retry", call);
+                }
+            }
```

Comment:
damn i had that in there at one time .... I was messing with `Void` vs. `void` and must have forgot to put it back ... that resulted in the test failure too.   Thanks for the catch.